source/slang/slang-ast-base.h

```diff
@@ -446,5 +532,115 @@ class Stmt : public ModifiableSyntaxNode
     void accept(IStmtVisitor* visitor, void* extra);
 };
 
+template<typename T>
+void DeclRef<T>::init(DeclRefBase* base)
+{
+    if (base && !Slang::as<T>(base->getDecl()))
+        declRefBase = nullptr;
+    else
+        declRefBase = base;
+}
+
+template<typename T>
+DeclRef<T>::DeclRef(Decl* decl)
+{
+    DeclRefBase* declRef = nullptr;
+    if (decl)
+    {
+        SLANG_ASSERT(decl->defaultDeclRef);
+        declRef = decl->defaultDeclRef;
+    }
+    else
+    {
+        declRef = nullptr;
+    }
```

Comment:
This side is a no-op, given that `declRef` is already initialized to null above, right?

Reply:
Fixed.